client/lib/i18n-utils/switch-locale.js

```diff
@@ -105,6 +107,33 @@ export async function switchCSS( elementId, cssUrl ) {
 	newLink.id = elementId;
 }
 
+function flipRTL( url, isRTL ) {
+	if ( isRTL ) {
+		return url.endsWith( '.rtl.css' ) ? url : url.replace( /\.css$/, '.rtl.css' );
+	}
+
+	return ! url.endsWith( '.rtl.css' ) ? url : url.replace( /\.rtl.css$/, '.css' );
+}
+
+function switchWebpackCSS( isRTL ) {
+	const currentLinks = document.querySelectorAll( 'link[rel="stylesheet"][data-webpack]' );
+
+	return map( currentLinks, async currentLink => {
+		const currentHref = currentLink.getAttribute( 'href' );
+		const newHref = flipRTL( currentHref, isRTL );
+		if ( currentHref === newHref ) {
+			return;
+		}
+
+		const newLink = await loadCSS( newHref, currentLink );
+		newLink.setAttribute( 'data-webpack', true );
+
+		if ( currentLink.parentElement ) {
+			currentLink.parentElement.removeChild( currentLink );
+		}
+	} );
+}
```

Comment:
There is some overlap in terms of functionality of this function with what we're doing in `DocumentHead` to [change the document title](https://github.com/Automattic/wp-calypso/blob/a7cc396c077724364135f72ddcf8f3ffbfd49e5d/client/components/data/document-head/index.jsx#L45-L47) on the client side when switching between different sections.

I wonder if we could extend `DocumentHead` to also cover the CSS link case, and then dispatch `setDocumentHeadLink` wen flipping betwwen LTR and RTL. (A word of caution, `DocumentHead`, and when to change or reset what has been a source of [subtle bugs](https://github.com/Automattic/wp-calypso/issues/24977), due to the many requirements there are for it -- amoung them SSR). However, it might be worth centralizing this kind of logic. (I don't want to block this PR on it though, can't wait to see this one at work :slightly_smiling_face: )

Reply:
I think I'll pass on this idea 🙂

`DocumentHead` owns the document title and some of the `meta` and `link` elements (but not all). The source of truth is Redux state, both in server and client rendering, and `DocumentHead` is a client-side tool that makes sure that the DOM is updated.

The script and stylesheet links, on the other hand, are not owned by Redux or any app state. They are owned by webpack. `setLocaleInDOM` updates these webpack-owned links in a careful cooperation with the webpack runtime.

I think these two very distinct concepts should be mixed together. The only thing they have in common is that the stylesheets live in the `head` element, that's all.

Reply:
Okay, fair enough.